Description: Make WinWStrDupFromWC endian agnostic
 WinWStrDupFromWC does not return a Windows wide character string
 (UTF-16 little endian) on big endian systems.
 .
 The following patch fixes this via FIX_ENDIAN_INT16.
 .
Forwarded: https://sourceforge.net/p/nsis/patches/273/
Author: Thomas Gaugler <thomas@dadie.net>
Last-Update: 2016-08-28

--- a/Source/winchar.cpp
+++ b/Source/winchar.cpp
@@ -111,7 +111,7 @@
   // NOTE: Anything outside the ASCII range will not convert correctly!
   size_t cch = wcslen(s);
   WINWCHAR* p = (WINWCHAR*) malloc(++cch * 2);
-  if (p) for (size_t i = 0; i < cch; ++i) p[i] = (unsigned char) s[i];
+  if (p) for (size_t i = 0; i < cch; ++i) p[i] = FIX_ENDIAN_INT16(s[i]);
   return p;
 #endif
 }
